README.adoc

```diff
@@ -0,0 +1,247 @@
+
+- It is *reactive*, because every request is non-blocking, asynchronous, and
+  wired up with the rest of the application using
+  http://www.reactive-streams.org[Reactive Streams].
+
+- It is *efficient*, because it works in batches, the basic unit of message
+  exchange between your driver and Pub/Sub. You `pull()` some, you `ack()` some.
+  One-message-at-a-time is an illusion created by drivers and incurs a
+  significant performance penalty along with operational complexity under the
+  hood.
+
+- It is *batteries-included*, because it provides goodies (out of the box
+  metrics integration, an adaptive rate limiter to help you avoid burning money
+  by continuously pulling and nack'ing messages when something is wrong with
+  your consumer, a `ScheduledExecutorService` implementation with a bounded task
+  queue to mitigate backpressure violating consumption) that assist real-world
+  production deployments.
+
+- It is *simple*, because there are 2 dozens of classes where half is used to
+  represent JSON models transmitted over the wire and the rest is just reactive
+  streams gluing. There are no smart retry mechanisms (though thanks to reactive
+  streams, you can introduce yours), there are no message lease time extending
+  background tasks (hence, you better consume your batch before it times out),
+  etc. Spend a minute on the source code and you are ready to write your own
+  Pub/Sub driver!
+
+Due to the constraints on the development resources (read as, _"it is just
+me"_), I needed to take some shortcuts to get this project going:
+
+- https://github.com/reactor/reactor-core/[Reactor Core] for Reactive Streams
+
+- https://github.com/reactor/reactor-netty[Reactor Netty] for HTTP requests
+
+- https://github.com/googleapis/google-api-java-client[Google APIs Client
+  Library]footnote:[This could have been replaced with a more lightweight
+  alternative, but given you have already been using Pub/Sub, it is highly
+  likely that you already sold your soul to some other Google Cloud services
+  too. Hence, no need to introduce an extra dependency.] for authentication
+
+- https://github.com/FasterXML/jackson-databind[Jackson]footnote:[https://github.com/googleapis/google-api-java-client[Google
+  APIs Client Library] already depends on this library.] for JSON serialization
+
+- http://micrometer.io/[Micrometer] for metrics
+
+== Getting Started
+
+You first need to add the `reactor-pubsub` artifact into your list of
+dependencies:
+
+```xml
+<dependency>
+    <groupId>com.vlkan</groupId>
+    <artifactId>reactor-pubsub</artifactId>
+    <version>${reactor-pubsub.version}</version>
+</dependency>
+```
+
+In order to instantiate a publisher and/or subscriber, you need to have certain
+basic components in place:
+
+- `ScheduledExecutorService` and `Scheduler` for scheduling asynchronous tasks
+- `PubsubAccessTokenCache` for authenticating HTTP requests
+- `MeterRegistry` for metrics
```

Comment:
I hope it is optional :) The preamble says that you need this component, hence the hope :)

Reply:
It isn't optional -- welcome to my opinionated project. But you are right, I will make it optional.

Reply:
I like the idea of having the metrics from day 1, but it may confuse some users that don't have/use MicroMeter (yet) :)